docs/examples/vector_stores/OpensearchDemo.ipynb

```diff
@@ -1,230 +1,233 @@
+   "metadata": {},
+   "source": [
+    "# Using as a vector index.\n",
+    "\n",
+    "Elasticsearch only supports Lucene indices, so only Opensearch is supported."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Note on setup**: We setup a local Opensearch instance through the following doc. https://opensearch.org/docs/1.0/\n",
+    "\n",
+    "If you run into SSL issues, try the following `docker run` command instead: \n",
+    "```\n",
+    "docker run -p 9200:9200 -p 9600:9600 -e \"discovery.type=single-node\" -e \"plugins.security.disabled=true\" opensearchproject/opensearch:1.0.1\n",
+    "```\n",
+    "\n",
+    "Reference: https://github.com/opensearch-project/OpenSearch/issues/1598"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
     },
-    "nbformat": 4,
-    "nbformat_minor": 4
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/jerryliu/Programming/gpt_index/.venv/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
+      "  from .autonotebook import tqdm as notebook_tqdm\n"
+     ]
+    }
+   ],
+   "source": [
+    "from os import getenv\n",
+    "from llama_index import SimpleDirectoryReader\n",
+    "from llama_index.vector_stores import OpensearchVectorStore, OpensearchVectorClient\n",
+    "from llama_index import GPTVectorStoreIndex\n",
+    "# http endpoint for your cluster (opensearch required for vector index usage)\n",
+    "endpoint = getenv(\"OPENSEARCH_ENDPOINT\", \"http://localhost:9200\")\n",
+    "# index to demonstrate the VectorStore impl\n",
+    "idx = getenv(\"OPENSEARCH_INDEX\", \"gpt-index-demo\")\n",
+    "# load some sample data\n",
+    "documents = SimpleDirectoryReader('../paul_graham_essay/data').load_data()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# OpensearchVectorClient stores text in this field by default\n",
+    "text_field = \"content\"\n",
+    "# OpensearchVectorClient stores embeddings in this field by default\n",
+    "embedding_field = \"embedding\"\n",
+    "# OpensearchVectorClient encapsulates logic for a\n",
+    "# single opensearch index with vector search enabled\n",
+    "client = OpensearchVectorClient(endpoint, idx, 1536, embedding_field=embedding_field, text_field=text_field)\n",
+    "# initialize vector store\n",
+    "vector_store = OpensearchVectorStore(client)\n",
+    "# initialize an index using our sample data and the client we just created\n",
+    "index = GPTVectorStoreIndex.from_documents(documents=documents)"
```

Comment:
@jerryjliu , would you please check if the code here would indeed persist the index to the OpenSearch?